datalad/distributed/ora_remote.py

```diff
@@ -167,9 +170,64 @@ def write_file(self, file_path, content, mode='w'):
 
         raise NotImplementedError
 
+    def make_readonly(self, path):
+        """Revoke write permissions for `path`
+
+        Parameters
+        ----------
+        path: Path or list of Path
+        """
+
+        # NOTE on VFAT: This will silently fail to remove write permission
+        # for the owner of a directory on VFAT, which implies that files
+        # still can be deleted w/o a "permission denied". The silence on that
+        # failure to have an effect is shared between `chmod` (via SSH) and
+        # `pathlib.Path.chmod` (local IO).
+        raise NotImplementedError
+
 
 class LocalIO(IOBase):
     """IO operation if the object tree is local (e.g. NFS-mounted)"""
+
+    @staticmethod
+    @contextmanager
+    def ensure_write_permission_parent(destination):
+        """Context manager to get permissions on directory and restore
+        afterwards
+
+        If git-annex ever touch this key store, the keys will
+        be in mode 444 directories, and we need to give permission
+        first.
+
+        Notes
+        -----
+
+        As pointed out in gh-5668, Ben is not convinced that this should really
+        be solved client-side, although it seems kind of fine, if we are able to
+        obtain the permission. However, store-side permissions should be set for
+        a reason, git-annex actually shouldn't touch the object tree and it
+        seems better to make sure that unaware people/programs don't
+        accidentally mess with a store's internal settings.
+        Hence, address the particular issue for the object tree only, not
+        enhancing the approach for all write operations ATM.
```

Comment:
This note is only relevant for a particular usage pattern in a particular context. Moreover, I think we should make it a habit of including personal message in documentation -- they can be placed in code comments.